homeassistant/components/wiz_light/light.py

```diff
@@ -0,0 +1,419 @@
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return light unique_id."""
+        return self._mac
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    async def async_turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        rgb = None
+        if ATTR_RGB_COLOR in kwargs:
+            rgb = kwargs.get(ATTR_RGB_COLOR)
+        if ATTR_HS_COLOR in kwargs:
+            rgb = color_utils.color_hs_to_RGB(
+                kwargs[ATTR_HS_COLOR][0], kwargs[ATTR_HS_COLOR][1]
+            )
+
+        brightness = None
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs.get(ATTR_BRIGHTNESS)
+
+        colortemp = None
+        if ATTR_COLOR_TEMP in kwargs:
+            kelvin = color_utils.color_temperature_mired_to_kelvin(
+                kwargs[ATTR_COLOR_TEMP]
+            )
+            colortemp = kelvin
+            _LOGGER.debug(
+                "[wizlight %s] kelvin changed and send to bulb: %s",
+                self._light.ip,
+                colortemp,
+            )
+
+        sceneid = None
+        if ATTR_EFFECT in kwargs:
+            sceneid = self._light.get_id_from_scene_name(kwargs[ATTR_EFFECT])
+
+        if sceneid == 1000:  # rhythm
+            pilot = PilotBuilder()
+        else:
+            pilot = PilotBuilder(
+                rgb=rgb, brightness=brightness, colortemp=colortemp, scene=sceneid
+            )
+        await self._light.turn_on(pilot)
+
+    async def async_turn_off(self, **kwargs):
+        """Instruct the light to turn off."""
+        await self._light.turn_off()
+
+    @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        return self._temperature
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        if self._bulbtype:
+            return self.kelvin_max_map()
+        # fallback
+        return color_utils.color_temperature_kelvin_to_mired(6500)
+
+    @property
+    def max_mireds(self):
+        """Return the warmest color_temp that this light supports."""
+        if self._bulbtype:
+            return self.kelvin_min_map()
+        # fallback
+        return color_utils.color_temperature_kelvin_to_mired(2500)
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        if self._bulbtype:
+            return self.featuremap()
+        # fallback
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT
+
+    @property
+    def effect(self):
+        """Return the current effect."""
+        return self._effect
+
+    @property
+    def effect_list(self):
+        """Return the list of supported effects."""
+        if self._bulbtype:
+            # Special filament bulb type
+            if self._bulbtype.name == "ESP56_SHTW3_01":
```

Comment:
Put these in a dict and then you can fetch the mapping

BULB_EFFECTS = {
   "ESP56_SHTW3_01": [8, 9, 14, 15, 17, 28, 29, 31],
   ...
}

Then you can do a dict get